examples/local-image/serveLocalGvmi.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Activity,
+  AgreementPoolService,
+  MarketService,
+  Package,
+  PaymentService,
+  WorkContext,
+  YagnaApi,
+  serveLocalGvmi,
+} from "@golem-sdk/golem-js";
+import { fileURLToPath } from "url";
+
+// get the absolute path to the local image in case this file is run from a different directory
+const getImagePath = (path: string) => fileURLToPath(new URL(path, import.meta.url).toString());
+
+const localImagePath = getImagePath("./alpine.gvmi");
+const server = serveLocalGvmi(localImagePath);
+
+async function main() {
+  console.log("Serving local image to the providers...");
+  await server.serve();
+  const { url, hash } = server.getImage();
+  const workload = Package.create({
+    imageHash: hash,
+    imageUrl: url,
+  });
+
+  console.log("Starting core services...");
+
+  const yagna = new YagnaApi();
+
+  const payment = new PaymentService(yagna);
+
+  await payment.run();
+
+  const allocation = await payment.createAllocation({
+    budget: 1.0,
+  });
+
+  const agreementPool = new AgreementPoolService(yagna);
+
+  const market = new MarketService(agreementPool, yagna);
+
+  await agreementPool.run();
+  await market.run(workload, allocation);
+  console.log("Core services started");
+
+  try {
+    const agreement = await agreementPool.getAgreement();
+    payment.acceptPayments(agreement);
+
+    const activity = await Activity.create(agreement, yagna);
+    // Stop listening for new proposals
+    await market.end();
+
+    const ctx = new WorkContext(activity, {});
```

Comment:
Just wondering, if we had an example where we were uploading files using `ctx.uploadFIle` or `ctx.downloadFile`, we would have to first create `StorageProvider` instance and do `init`, then pass it to the `WorkContext` constructor.. 
and in this case we would have two instances of the `StorageProvider` server. So maybe this `serveLocalGvmi` helper should take `StorageProvider?` as a dependency and only create it when it is null?